src/renderer.rs

```diff
@@ -412,39 +412,41 @@ impl TextureBuffer {
     }
 
     /// Write an RGBA buffer at the specified location.
-    pub fn write_rgba_at(&mut self, buffer: &[u8], width: usize, pos: (usize, usize)) {
-        for row in 0..buffer.len() / width {
-            let dst_start = (pos.1 + row) * self.width + pos.0 * 4;
-            if dst_start >= self.inner.len() {
-                break;
-            }
-            let dst_row = &mut self.inner[dst_start..];
-
-            let src_start = row * width;
-            let src_row = &buffer[src_start..src_start + cmp::min(width, dst_row.len())];
-
-            let pixels = src_row.chunks(4).enumerate().filter(|(_i, pixel)| pixel != &[0, 0, 0, 0]);
-            for (i, pixel) in pixels {
-                dst_row[i * 4..i * 4 + 4].copy_from_slice(pixel)
-            }
-        }
+    pub fn write_rgba_at(&mut self, buffer: &[u8], width: usize, pos: (isize, isize)) {
+        self.write_buffer_inner::<4>(buffer, width, pos)
     }
 
     /// Write an RGB buffer at the specified location.
-    pub fn write_rgb_at(&mut self, buffer: &[u8], width: usize, pos: (usize, usize)) {
+    pub fn write_rgb_at(&mut self, buffer: &[u8], width: usize, pos: (isize, isize)) {
+        self.write_buffer_inner::<3>(buffer, width, pos);
+    }
+
+    fn write_buffer_inner<const N: usize>(
+        &mut self,
+        buffer: &[u8],
+        width: usize,
+        pos: (isize, isize),
+    ) {
         for row in 0..buffer.len() / width {
-            let dst_start = (pos.1 + row) * self.width + pos.0 * 4;
-            if dst_start >= self.inner.len() {
+            let dst_start = (pos.1 + row as isize) * self.width as isize + pos.0 * 4;
```

Comment:
If the X pos is `-1`, but Y is positive, wouldn't this cause the `dst_start` to be in the previous row and thus not cause it to be cut off but instead cause it to be rendered at the wrong location?

Reply:
Aye, I see, should just skip on negative `pos.0` I guess?

Reply:
I don't think that works, because then you'd be skipping the entire glyph?

Reply:
I've already send a patch that does it differently.